File: java-vertx/src/main/java/io/swagger/server/api/model/V3DeparturesSpecificParameters.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.ArrayList;
import java.util.List;
import org.threeten.bp.OffsetDateTime;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3DeparturesSpecificParameters
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3DeparturesSpecificParameters   {
  private Integer directionId = null;

  private Boolean gtfs = null;

  private OffsetDateTime dateUtc = null;

  private Integer maxResults = null;

  private Boolean includeCancelled = null;

  private Boolean lookBackwards = null;

  /**
   * Gets or Sets expand
   */
  public enum ExpandEnum {
    ALL("All"),
    STOP("Stop"),
    ROUTE("Route"),
    RUN("Run"),
    DIRECTION("Direction"),
    DISRUPTION("Disruption"),
    VEHICLEDESCRIPTOR("VehicleDescriptor"),
    VEHICLEPOSITION("VehiclePosition"),
    NONE("None");

    private String value;

    ExpandEnum(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static ExpandEnum fromValue(String value) {
      for (ExpandEnum b : ExpandEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      return null;
    }

  }
  
  private List<ExpandEnum> expand = null;

  private Boolean includeGeopath = null;

  public V3DeparturesSpecificParameters directionId(Integer directionId) {
    this.directionId = directionId;
    return this;
  }

  /**
   * Filter by identifier of direction of travel; values returned by Directions API - /v3/directions/route/{route_id}
   * @return directionId
   **/
    public Integer getDirectionId() {
    return directionId;
  }

  public void setDirectionId(Integer directionId) {
    this.directionId = directionId;
  }

  public V3DeparturesSpecificParameters gtfs(Boolean gtfs) {
    this.gtfs = gtfs;
    return this;
  }

  /**
   * Indicates that stop_id parameter will accept \"GTFS stop_id\" data
   * @return gtfs
   **/
    public Boolean isGtfs() {
    return gtfs;
  }

  public void setGtfs(Boolean gtfs) {
    this.gtfs = gtfs;
  }

  public V3DeparturesSpecificParameters dateUtc(OffsetDateTime dateUtc) {
    this.dateUtc = dateUtc;
    return this;
  }

  /**
   * Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)
   * @return dateUtc
   **/
    public OffsetDateTime getDateUtc() {
    return dateUtc;
  }

  public void setDateUtc(OffsetDateTime dateUtc) {
    this.dateUtc = dateUtc;
  }

  public V3DeparturesSpecificParameters maxResults(Integer maxResults) {
    this.maxResults = maxResults;
    return this;
  }

  /**
   * Maximum number of results returned
   * @return maxResults
   **/
    public Integer getMaxResults() {
    return maxResults;
  }

  public void setMaxResults(Integer maxResults) {
    this.maxResults = maxResults;
  }

  public V3DeparturesSpecificParameters includeCancelled(Boolean includeCancelled) {
    this.includeCancelled = includeCancelled;
    return this;
  }

  /**
   * Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only
   * @return includeCancelled
   **/
    public Boolean isIncludeCancelled() {
    return includeCancelled;
  }

  public void setIncludeCancelled(Boolean includeCancelled) {
    this.includeCancelled = includeCancelled;
  }

  public V3DeparturesSpecificParameters lookBackwards(Boolean lookBackwards) {
    this.lookBackwards = lookBackwards;
    return this;
  }

  /**
   * Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.
   * @return lookBackwards
   **/
    public Boolean isLookBackwards() {
    return lookBackwards;
  }

  public void setLookBackwards(Boolean lookBackwards) {
    this.lookBackwards = lookBackwards;
  }

  public V3DeparturesSpecificParameters expand(List<ExpandEnum> expand) {
    this.expand = expand;
    return this;
  }

  public V3DeparturesSpecificParameters addExpandItem(ExpandEnum expandItem) {
    if (this.expand == null) {
      this.expand = new ArrayList<ExpandEnum>();
    }
    this.expand.add(expandItem);
    return this;
  }

  /**
   * List of objects to be returned in full (i.e. expanded) - options include: All, Stop, Route, Run, Direction, Disruption, VehiclePosition, VehicleDescriptor or None.  Run must be expanded to receive VehiclePosition and VehicleDescriptor information.
   * @return expand
   **/
    public List<ExpandEnum> getExpand() {
    return expand;
  }

  public void setExpand(List<ExpandEnum> expand) {
    this.expand = expand;
  }

  public V3DeparturesSpecificParameters includeGeopath(Boolean includeGeopath) {
    this.includeGeopath = includeGeopath;
    return this;
  }

  /**
   * Indicates if the route geopath should be returned
   * @return includeGeopath
   **/
    public Boolean isIncludeGeopath() {
    return includeGeopath;
  }

  public void setIncludeGeopath(Boolean includeGeopath) {
    this.includeGeopath = includeGeopath;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3DeparturesSpecificParameters v3DeparturesSpecificParameters = (V3DeparturesSpecificParameters) o;
    return Objects.equals(this.directionId, v3DeparturesSpecificParameters.directionId) &&
        Objects.equals(this.gtfs, v3DeparturesSpecificParameters.gtfs) &&
        Objects.equals(this.dateUtc, v3DeparturesSpecificParameters.dateUtc) &&
        Objects.equals(this.maxResults, v3DeparturesSpecificParameters.maxResults) &&
        Objects.equals(this.includeCancelled, v3DeparturesSpecificParameters.includeCancelled) &&
        Objects.equals(this.lookBackwards, v3DeparturesSpecificParameters.lookBackwards) &&
        Objects.equals(this.expand, v3DeparturesSpecificParameters.expand) &&
        Objects.equals(this.includeGeopath, v3DeparturesSpecificParameters.includeGeopath);
  }

  @Override
  public int hashCode() {
    return Objects.hash(directionId, gtfs, dateUtc, maxResults, includeCancelled, lookBackwards, expand, includeGeopath);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3DeparturesSpecificParameters {\n");
    
    sb.append("    directionId: ").append(toIndentedString(directionId)).append("\n");
    sb.append("    gtfs: ").append(toIndentedString(gtfs)).append("\n");
    sb.append("    dateUtc: ").append(toIndentedString(dateUtc)).append("\n");
    sb.append("    maxResults: ").append(toIndentedString(maxResults)).append("\n");
    sb.append("    includeCancelled: ").append(toIndentedString(includeCancelled)).append("\n");
    sb.append("    lookBackwards: ").append(toIndentedString(lookBackwards)).append("\n");
    sb.append("    expand: ").append(toIndentedString(expand)).append("\n");
    sb.append("    includeGeopath: ").append(toIndentedString(includeGeopath)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3DeparturesSpecificParameters.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3DeparturesSpecificParameters {
    /// <summary>
    /// Filter by identifier of direction of travel; values returned by Directions API - /v3/directions/route/{route_id}
    /// </summary>
    /// <value>Filter by identifier of direction of travel; values returned by Directions API - /v3/directions/route/{route_id}</value>
    [DataMember(Name="direction_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_id")]
    public int? DirectionId { get; set; }

    /// <summary>
    /// Indicates that stop_id parameter will accept \"GTFS stop_id\" data
    /// </summary>
    /// <value>Indicates that stop_id parameter will accept \"GTFS stop_id\" data</value>
    [DataMember(Name="gtfs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "gtfs")]
    public bool? Gtfs { get; set; }

    /// <summary>
    /// Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)
    /// </summary>
    /// <value>Filter by the date and time of the request (ISO 8601 UTC format) (default = current date and time)</value>
    [DataMember(Name="date_utc", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "date_utc")]
    public DateTime? DateUtc { get; set; }

    /// <summary>
    /// Maximum number of results returned
    /// </summary>
    /// <value>Maximum number of results returned</value>
    [DataMember(Name="max_results", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "max_results")]
    public int? MaxResults { get; set; }

    /// <summary>
    /// Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only
    /// </summary>
    /// <value>Indicates if cancelled services (if they exist) are returned (default = false) - metropolitan train only</value>
    [DataMember(Name="include_cancelled", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "include_cancelled")]
    public bool? IncludeCancelled { get; set; }

    /// <summary>
    /// Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.
    /// </summary>
    /// <value>Indicates if filtering runs (and their departures) to those that arrive at destination before date_utc (default = false). Requires max_results &gt; 0.</value>
    [DataMember(Name="look_backwards", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "look_backwards")]
    public bool? LookBackwards { get; set; }

    /// <summary>
    /// List of objects to be returned in full (i.e. expanded) - options include: All, Stop, Route, Run, Direction, Disruption, VehiclePosition, VehicleDescriptor or None.  Run must be expanded to receive VehiclePosition and VehicleDescriptor information.
    /// </summary>
    /// <value>List of objects to be returned in full (i.e. expanded) - options include: All, Stop, Route, Run, Direction, Disruption, VehiclePosition, VehicleDescriptor or None.  Run must be expanded to receive VehiclePosition and VehicleDescriptor information.</value>
    [DataMember(Name="expand", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "expand")]
    public List<string> Expand { get; set; }

    /// <summary>
    /// Indicates if the route geopath should be returned
    /// </summary>
    /// <value>Indicates if the route geopath should be returned</value>
    [DataMember(Name="include_geopath", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "include_geopath")]
    public bool? IncludeGeopath { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3DeparturesSpecificParameters {\n");
      sb.Append("  DirectionId: ").Append(DirectionId).Append("\n");
      sb.Append("  Gtfs: ").Append(Gtfs).Append("\n");
      sb.Append("  DateUtc: ").Append(DateUtc).Append("\n");
      sb.Append("  MaxResults: ").Append(MaxResults).Append("\n");
      sb.Append("  IncludeCancelled: ").Append(IncludeCancelled).Append("\n");
      sb.Append("  LookBackwards: ").Append(LookBackwards).Append("\n");
      sb.Append("  Expand: ").Append(Expand).Append("\n");
      sb.Append("  IncludeGeopath: ").Append(IncludeGeopath).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
